### PR TITLE
Document update for v8

### PR DESCRIPTION
**Pull request by doogstar**
In the spirit of #1853 but I don't think specifically mentioned on there.

Hopefully I've done it better this time by combining the old as v7 and the new with updates into the one pull request. :)

### CODE REVIEW

**Comment by sofietoft** [comment]
@doogstar - apologies for the delay here! Still working on getting this reviewed 💪 

As we're closing Hacktoberfest, I've gone ahead and added the Hacktoberfest label to this PR 😄

**Comment by doogstar** [comment]
Thanks @sofietoft no rush, hopefully all goes through OK once you get a chance to review :)

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+## Preparations
+
+Create a new ASP.NET or class project in Visual Studio add references to the Umbraco.Forms.Core.dll.
+
+## Adding the type to Forms
+
+The Forms API contains a collection of classes that the provider model automatically registers. So to add a new type to Forms you inherit from the right class. In the sample below we use the class for the workflow type.
+
+```csharp
+public class Class1 : Umbraco.Forms.Core.WorkflowType
```
Comment (warrenbuckley) [inline Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md]: Can we have a better name please than `Class1` @doogstar

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+When you implement this class you get two methods added. One of them is Execute which performs the execution of the workflow and the other is a method which validates the workflow settings, we will get back to these settings later on.
+
+Even though we have the class inheritance in place, we still need to add a bit of default information.
+
+## Setting up basic type information
+
+Even though we have the class inheritance in place, we still need to add a bit of default information. This information is added in the class's empty constructor like this:
+
+```csharp
+public Class1() {
```
Comment (warrenbuckley) [inline Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md]: Again please change the name, maybe call it `LogWorkflow` or similar?

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+With the attribute in place, the property value is set every time the class is instantiated by Umbraco Forms. This means you can use the property in your code like this:
+
+```csharp
+[Umbraco.Forms.Core.Attributes.Setting("Document ID",
+        description = "Node the log entry belongs to",
+        view = "Pickers.Content")]
+public string Document { get; set; }
+
+public override Enums.WorkflowExecutionStatus Execute(Record record) {
```
Comment (warrenbuckley) [inline Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md]: Tiny detail as your being specific with class names in your examples, make sure the one on line 20 is the same.

**Comment by warrenbuckley** [comment]
Great stuff @doogstar and thanks for the PR
Super tiny/trivial stuff to fix up then think its good for @sofietoft to merge in once they are done :)

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md (hunk 1 above)
Comment (doogstar) [reply Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md]: Renamed as suggested, sorry! :)

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md (hunk 2 above)
Comment (doogstar) [reply Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md]: Yep great suggestion

File: Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md (hunk 3 above)
Comment (doogstar) [reply Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md]: Thanks - made both the full version (especially with the differnce between v7 & v8 and user can decide to leave as is or change to a using statement

**Comment by warrenbuckley** [comment]
@Sofietoft I’m happy so over to you, in case you bed/want any final bits done

**Comment by sofietoft** [comment]
Thanks for all the help @doogstar and @warrenbuckley 😄 
I'll go ahead and get this one merged in!

**Pull request merged**